src/main.c

```diff
@@ -479,11 +514,20 @@ static const char *listener_setup_ssl_picotls(struct listener_config_t *listener
     X509 *cert;
     STACK_OF(X509) * cert_chain;
     int ret;
+#if OPENSSL_VERSION_NUMBER >= 0x10101000L
+    ptls_cipher_suite_t **cipher_suites = build_picotls_cipher_suite(ssl_ctx);
+    if (!cipher_suites) {
+        free(pctx);
+        return "Failed to setup cipher suites";
+    }
+#else
+    ptls_cipher_suite_t **cipher_suites = ptls_openssl_cipher_suites;
+#endif
```

Comment:
I think we need to handle the arguments of `cipher-suites-tls1.3` directly for picotls. Otherwise, users would not be able to customize the cipher-suites when using OpenSSL older than 1.1.0.

Reply:
Would you move the parsing of the cipher description strings to picotls, or keep them in `src/main.c` in h2o?

Reply:
I think we can keep it outside of picotls.

It's the application's responsibility to build the cipher-suites array (`ptls_context_t::cipher_suites`), including allocating sufficient memory for holding the list. I do not think there is a good way of providing the parsing function from inside of picotls considering these requirements.